docs/DeveloperGuide.md

```diff
@@ -1,34 +1,152 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 3. Implementation
+### 3.1 [Proposed] Features
+#### 3.1.1 List By Category
+##### 3.1.1.1 Proposed Implementation
+The list by category mechanism is facilitated by ListCommand which extends Command.
 
+Given below is an example usage scenario and how the mechanism behaves at each step.
 
-## Product Scope
-### Target user profile
+Step1: The user adds a task:  `add n/2113 c/DEADLINE d/2020-03-30 t/11:59` to add a task.
+Since the user adds a new category `DEADLINE`, the categoryMap wil add one more key whose value is the category and the value is
+the index for this task.
 
-{Describe the target user profile}
+Step2: The users adds a task: `add n/3145 c/CLASS` to add a class. 
+Since the user adds a new category `CLASS`, the categoryMap wil add one more key whose value is the category and the value is
+the index for this task.
 
-### Value proposition
+Step3: Users list the class he just added by category CLASS: `list c/CLASS`. The TaskList will
+return the index of the tasks based on the categoryMap. Then the task in the `CLASS` category
+will be displayed. 
 
-{Describe the value proposition: what problem does it solve?}
+This is the Sequence Diagram for list by category. 
+![Sequence Diagram for View](images/listCategorySequence.png)
+This is the UML design for list by category.
+![UML for View](images/listCategory.png)
+##### 3.1.1.2 Design Considerations
+Aspect: How to find certain category. 
 
-## User Stories
+Alternative 1 (current choice): store HashMap to map category with the key. 
+
+Alternative 2 (easy to do): linear search when searching tasks. 
+
+
+#### 3.1.2 [Proposed] View month
+##### 3.1.2.1 Proposed Implementation
+The view month mechanism is facilitated by CalendarCommand which extends Command.
+
+Given below is an example usage scenario and how the mechanism behaves at each step.
+
+1. The user inputs the command word `calendar`. Upon which, the instance of parser will return a CalendarCommand for execution
+1. CalendarCommand initialises with the following variables, with the help from a calendar class containing the necessary methods related to day/date.
+    * month - if user does not input month, it uses computer's current month
+    * startingDay - which day of the week the first day of the months begins on (0-6, where 0 is Monday)
+    * totalDays - how many days in that month
+    * totalWeeks - number of weeks of the month
+  
+1. The `calendar` command calls on TaskList#findTaskDate() for each day of the month to generate the task listing for a particular day. Only the task event and description is added to calendar view. 
+1. The calendar generated is then stored in a string and UI#showUserMesssage() is called to display the calendar. 
+
+The class diagram below shows the relationships between the different classes required by the `calendar` feature.
```

Comment:
Should the abstract keyword for abstract classes in the class diagram be above the class name as seen in the textbook?